pandas/core/groupby/groupby.py

```diff
@@ -2696,6 +2697,103 @@ def _reindex_output(
 
         return output.reset_index(drop=True)
 
+    def sample(
+        self,
+        n: Optional[int] = None,
+        frac: Optional[float] = None,
+        replace: bool = False,
+        weights: Optional[Union[Sequence, Series]] = None,
+        random_state=None,
+    ):
+        """
+        Return a random sample of items from each group.
+
+        You can use `random_state` for reproducibility.
+
+        .. versionadded:: 1.1.0
+
+        Parameters
+        ----------
+        n : int, optional
+            Number of items to return. Cannot be used with `frac`.
+            Default = 1 if `frac` is None.
+        frac : float, optional
+            Fraction of items to return. Cannot be used with `n`.
+        replace : bool, default False
+            Allow or disallow sampling of the same row more than once.
+        weights : list-like, optional
+            Default None results in equal probability weighting.
+            If passed a list-like then values must have the same length as
+            the underlying object and will be used as sampling probabilities
+            after normalization within each group.
+        random_state : int, array-like, BitGenerator, np.random.RandomState, optional
+            If int, array-like, or BitGenerator (NumPy>=1.17), seed for
```

Comment:
It it is a BitGenerator, do you use a Generator to produce the random samples or a RandomState.  Best practice is to use a Generator since RandomState is effectively frozen in time. If an int, it is used as a seed for `np.random.default_rng()` or `RandomState` if NumPy >= 1.17?

Reply:
This is following a pattern similar to the one used in pandas.core.generic.sample of processing the random_state according to pandas.core.common.random_state:

https://github.com/pandas-dev/pandas/blob/c71bfc36211b5e2d860a06d8fbef902b757bd6e4/pandas/core/common.py#L394